pageserver/src/layered_repository.rs

```diff
@@ -1080,12 +1120,14 @@ struct GcInfo {
     /// last-record LSN
     ///
     /// FIXME: is this inclusive or exclusive?
-    cutoff: Lsn,
+    horizon_cutoff: Lsn,
 
-    /// In addition to 'retain_lsns', keep everything newer than 'SystemTime::now()'
-    /// minus 'pitr_interval'
+    /// In addition to 'retain_lsns' and 'horizon_cutoff', keep everything newer than this
+    /// point.
     ///
-    pitr: Duration,
+    /// This is calculated by finding a number such that a record is needed for PITR
+    /// if only if its LSN is larger than 'pitr_cutoff'.
+    pitr_cutoff: Lsn,
 }
```

Comment:
You can combine `horizon_cutoff` and `pitr_cutoff` into one value. The code in `gc` prints different debug messages, and uses separate counters depending on which one was smaller, but I don't think that distinction is needed. Instead, you could just print a debug message in `update_gc_info`, to indicate which was dominant.

Reply:
Yeah. However, it's not a trivial change though. I think it's better to leave that to a separate PR/ follow-up